contracts/ArbitrableDirectory.sol

```diff
@@ -0,0 +1,827 @@
+            "Directory: The organization has wrong status."
+        );
+        (,,,,,,, address orgOwner, address director,, bool isDirectorshipAccepted) = orgId.getOrganization(_organization);
+        require(orgOwner == msg.sender || (director == msg.sender && isDirectorshipAccepted), "Directory: Only organization owner or director can request a withdrawal.");
+
+        organization.withdrawalRequestTime = now;
+        organization.status = Status.WithdrawalRequested;
+        uint index = organizationsIndex[_organization];
+        if (index != 0) {
+            bytes32 lastOrg = registeredOrganizations[registeredOrganizations.length - 1];
+            registeredOrganizations[index] = lastOrg;
+            organizationsIndex[lastOrg] = index;
+            registeredOrganizations.length--;
+            organizationsIndex[_organization] = 0;
+            emit OrganizationRemoved(_organization);
+        }
+    }
+
+    /** @dev Withdraw all the Lif tokens deposited when the request was made.
+     *  @param _organization The ID of the organization to un-register.
+     */
+    function withdrawTokens(bytes32 _organization) external {
+        Organization storage organization = organizationData[_organization];
+        require(
+            organization.status == Status.WithdrawalRequested,
+            "Directory: The organization has wrong status."
+        );
+        require(now - organization.withdrawalRequestTime > withdrawTimeout, "Directory: Tokens can only be withdrawn after the timeout.");
+        (,,,,,,, address orgOwner,,,) = orgId.getOrganization(_organization);
+        organization.status = Status.Absent;
+        organization.withdrawalRequestTime = 0;
+        uint stake = organization.lifStake;
+        organization.lifStake = 0;
+        require(lif.transfer(orgOwner, stake), "Directory: The token transfer must not fail.");
+    }
+
+    /** @dev Give a ruling for a dispute. Can only be called by the arbitrator. TRUSTED.
+     *  Accounts for the situation where the winner loses a case due to paying less appeal fees than expected.
+     *  @param _disputeID ID of the dispute in the arbitrator contract.
+     *  @param _ruling Ruling given by the arbitrator. Note that 0 is reserved for "Refused to arbitrate".
+     */
+    function rule(uint _disputeID, uint _ruling) public {
+        Party resultRuling = Party(_ruling);
+        bytes32 organizationID = arbitratorDisputeIDToOrg[msg.sender][_disputeID];
+        Organization storage organization = organizationData[organizationID];
+
+        Challenge storage challenge = organization.challenges[organization.challenges.length - 1];
+        Round storage round = challenge.rounds[challenge.rounds.length - 1];
+        require(_ruling <= RULING_OPTIONS, "Directory: Invalid ruling option.");
+        require(address(challenge.arbitrator) == msg.sender, "Directory: Only the arbitrator can give a ruling.");
```

Comment:
it makes sense to move this condition to the top

Reply:
Which one? Arbitrator, Ruling_Options or both?

Reply:
lines 562, 563

Reply:
I could only move L562, since L563 is tied to `challenge` declaration